router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -0,0 +1,546 @@
+	}
+	// Update the refreshed account information into in-memory map(cache)
+	authErrHandler.destAuthInfoMap[refTokenParams.AccountId] = &AuthResponse{
+		Account: accountSecret,
+	}
+	authStats.statName = fmt.Sprintf("%s_success", refTokenParams.EventNamePrefix)
+	authStats.errorMessage = ""
+	authStats.SendCountStat()
+	authErrHandler.logger.Debugf("[%s request] :: (Write) %s response received(rt-worker-%d): %s\n", loggerNm, logTypeName, refTokenParams.WorkerId, response)
+	return http.StatusOK
+}
+
+func getRefreshTokenErrResp(response string, accountSecret *AccountSecret) (message string) {
+	if err := json.Unmarshal([]byte(response), &accountSecret); err != nil {
+		// Some problem with AccountSecret unmarshalling
+		message = err.Error()
+	} else if gjson.Get(response, "body.code").String() == INVALID_REFRESH_TOKEN_GRANT {
+		message = INVALID_REFRESH_TOKEN_GRANT
+	} else if !router_utils.IsNotEmptyString(accountSecret.AccessToken) {
+		// Status is 200, but no accesstoken is sent
+		message = `Empty Token cannot be processed further`
+	}
+	return message
+}
+
+func (authStats *OAuthStats) SendTimerStats(startTime time.Time) {
+	stats.NewTaggedStat(authStats.statName, stats.TimerType, stats.Tags{
+		"id":              authStats.id,
+		"workspaceId":     authStats.workspaceId,
+		"rudderCategory":  authStats.rudderCategory,
+		"isCallToCpApi":   strconv.FormatBool(authStats.isCallToCpApi),
+		"authErrCategory": authStats.authErrCategory,
+		"destDefName":     authStats.destDefName,
+	}).SendTiming(time.Since(startTime))
+}
+
+// Send count type stats related to OAuth(Destination)
+func (refStats *OAuthStats) SendCountStat() {
+	stats.NewTaggedStat(refStats.statName, stats.CountType, stats.Tags{
+		"id":              refStats.id,
+		"workspaceId":     refStats.workspaceId,
+		"rudderCategory":  refStats.rudderCategory,
+		"errorMessage":    refStats.errorMessage,
+		"isCallToCpApi":   strconv.FormatBool(refStats.isCallToCpApi),
+		"authErrCategory": refStats.authErrCategory,
+		"destDefName":     refStats.destDefName,
+		"isTokenFetch":    strconv.FormatBool(refStats.isTokenFetch),
+	}).Increment()
+}
+
+func (authErrHandler *OAuthErrResHandler) DisableDestination(destination backendconfig.DestinationT, workspaceId string) (statusCode int, respBody string) {
+	authErrHandlerTimeStart := time.Now()
+	destinationId := destination.ID
+	resMgrErr := authErrHandler.NewMutex(destinationId, DISABLE_DEST)
+	if resMgrErr != nil {
+		panic(resMgrErr)
+	}
+
+	disableDestStats := &OAuthStats{
+		id:              destinationId,
+		workspaceId:     workspaceId,
+		rudderCategory:  "destination",
+		statName:        "",
+		isCallToCpApi:   false,
+		authErrCategory: DISABLE_DEST,
+		errorMessage:    "",
+		destDefName:     destination.DestinationDefinition.Name,
+	}
+	defer func() {
+		disableDestStats.statName = "disable_destination_total_req_latency"
+		disableDestStats.isCallToCpApi = false
+		disableDestStats.SendTimerStats(authErrHandlerTimeStart)
```

Comment:
[minor] This part of code here concerns me a bit, in terms of readability and accidents. Having to change the properties of a struct just before you call a method of it, can cause confusion and issues. 

Is there a way to avoid doing this? For example, what if:
```suggestion
		disableDestStats.SendTimerStats("disable_destination_total_req_latency", authErrHandlerTimeStart, false)
```

Or even better create one method for `isCallToCpApi=true` and one for `isCallToCpApi=false`